source/core.py

```diff
@@ -250,6 +250,48 @@ def onEndSession(evt):
 	import windowUtils
 	class MessageWindow(windowUtils.CustomWindow):
 		className = u"wxWindowClassNR"
+		batteryCache = None
+		REASON_INITIALIZE = 1
+		def __init__(self, windowName=None):
+			super(MessageWindow, self).__init__(windowName)
+			self.say_battery_status(reason=self.REASON_INITIALIZE)
+	
+		def windowProc(self, hwnd, msg, wParam, lParam):
+			import ui
+			#Just define these constants here, so we don't have to import win32con
+			WM_POWERBROADCAST = 0x218
+			WM_DISPLAYCHANGE = 0x7e
+			PBT_APMPOWERSTATUSCHANGE = 0xA
+			if msg == WM_POWERBROADCAST and wParam == PBT_APMPOWERSTATUSCHANGE:
+				self.say_battery_status()
+			# Resolution detection comes from an article found at https://msdn.microsoft.com/en-us/library/ms812142.aspx.
+			elif msg == WM_DISPLAYCHANGE:
+				ui.message("Landscape" if lParam%0x10000 > lParam/0x10000 else "Portrait")

```

Comment:
These strings need to be made translatable, including translator comments. That requires that both strings are on separate lines so the translator comment can apply to the correct string.